src/components/FileUpload.js

```diff
@@ -0,0 +1,51 @@
+import React, { useState } from "react";
+import axios from "axios";
+
+export default function FileUpload() {
+  const [pdfFile, setPdfFile] = useState(null);
+
+  const fileType = ["application/pdf"];
+
+  const handlePdfFileChange = (e) => {
+    let selectedFile = e.target.files[0];
+    if (selectedFile) {
+      if (selectedFile && fileType.includes(selectedFile.type)) {
+        let reader = new FileReader();
+        reader.readAsDataURL(selectedFile);
+        reader.onloadend = (e) => {
+          setPdfFile(selectedFile);
+        };
+      }
+    } else {
+      setPdfFile(null);
+    }
+  };
+
+  const handlePdfSubmit = (e) => {
+    e.preventDefault();
+    if (pdfFile !== null) {
+      const data = new FormData();
+      data.append("file", pdfFile);
+
+      axios
+        .post("http://localhost:8000/api/v1/file/upload", data, {})
```

Comment:
Don't use magic numbers/string for routes especially. 
Create path.js file where you put constants for endpoints.
Example:
const basePath = "http://localhost:8000/api/v1/file"
const uploadPath = "/upload"

and then append the string accordingly when using axios.
Also create util directory and configure axios request there.